### PR TITLE
Allow sentry user to control resolution of captured Flutter screenshots

### LINKED ISSUE #1200: Allow sentry user to control resolution of captured Flutter screenshots
Feature request: Allow Sentry users to control the resolution at which Flutter [screenshots are captured by Sentry](https://docs.sentry.io/platforms/flutter/enriching-events/screenshots/), allow the Sentry user to control the resolution at which screenshots are captured.

For example, Sentry's Unity SDK currently allows users to select among:
- as is
- high - 1920p
- medium - 720p
- low - 420p

The request is to allow similar choice in resolution for Flutter screenshots. 

- There is concern that larger-resolution screenshots may cause bandwidth problems for events captured on low-end mobile devices and regions with poor cellular connectivity. The thought is that being able to adjust resolution would help with this.

┆Issue is synchronized with this [Jira Improvement](https://getsentry.atlassian.net/browse/FEEDBACK-1695) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Introduce `SentryscreenshotQulity`
- Render screenshots according to quality
- We now also support full resolution instead of just rendering logical pixels

Full Quality:
Bildschirm­foto 2023-02-20 um 17 10 12

Small Quality:
Bildschirm­foto 2023-02-20 um 17 15 08

## :bulb: Motivation and Context

Closes #1200

## :green_heart: How did you test it?

Ran different resolutions with sample app.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/lib/src/screenshot/sentry_screenshot_quality.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+enum SentryScreenshotQuality {
```
Comment (marandaneto) [inline flutter/lib/src/screenshot/sentry_screenshot_quality.dart:2]:
```suggestion
/// The quality of the attached screenshot
enum SentryScreenshotQuality {
```

File: flutter/lib/src/screenshot/sentry_screenshot_quality.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  int? targetResolution() {
+    switch (this) {
+      case SentryScreenshotQuality.full:
+        return null; // Keep current scale
+      case SentryScreenshotQuality.high:
+        return 1920;
+      case SentryScreenshotQuality.medium:
+        return 1280;
+      case SentryScreenshotQuality.low:
+        return 854;
```
Comment (marandaneto) [inline flutter/lib/src/screenshot/sentry_screenshot_quality.dart:17]: Did you come up with the numbers from somewhere? are they standard mobile resolutions?

Comment (marandaneto) [reply flutter/lib/src/screenshot/sentry_screenshot_quality.dart:17]: NVM asked @bitsandfoxes since its the same on Unity.

**Comment by marandaneto** [comment]
@denrase can we add testing for the screen quality?

File: flutter/lib/src/screenshot/sentry_screenshot_quality.dart (hunk 2 above)
Comment (denrase) [reply flutter/lib/src/screenshot/sentry_screenshot_quality.dart:17]: Yeah i took the values from there, assuming they are sane defaults, as they are already in use.

**Comment by denrase** [comment]
@marandaneto Will check if we can add an integration test.

**Review by marandaneto — APPROVED** [review]
Thanks @denrase

**Pull request merged**